core/src/autogluon/core/_setup_utils.py

```diff
@@ -17,13 +17,13 @@
 # Only put packages here that would otherwise appear multiple times across different module's setup.py files.
 DEPENDENT_PACKAGES = {
     "boto3": ">=1.10,<2",  # <2 because unlikely to introduce breaking changes in minor releases. >=1.10 because 1.10 is 3 years old, no need to support older
-    "numpy": ">=1.25.0,<2.1.4",  # "<{N+3}" upper cap, where N is the latest released minor version, assuming no warnings using N
+    "numpy": ">=1.25.0,<2.3.0",  # "<{N+3}" upper cap, where N is the latest released minor version, assuming no warnings using N
```

Comment:
Just a note that `numpy==1.26.x` is getting installed currently due to catboost cap https://github.com/catboost/catboost/issues/2671

Reply:
The CatBoost support for NP 2.0 isn't released yet, so we'll leave it as is for now and update once the CatBoost release is complete.